packages/health-gorilla-core/src/aoe.ts

```diff
@@ -51,24 +50,16 @@ export function getMissingRequiredQuestionnaireItems(
           'https://www.healthgorilla.com/fhir/StructureDefinition/questionnaire-requiredwhenspecimen'
         ) === true)
     ) {
-      required.push(item.linkId);
-
       const answers = answersByLinkId?.[item.linkId];
       if (answers && answers.length > 0) {
-        let hasValue = false;
         for (const answer of answers) {
           const values = arrayify(
             getTypedPropertyValue({ type: 'QuestionnaireResponseItemAnswer', value: answer }, 'value[x]')
           );
           if (values?.some((v) => v.value)) {
-            hasValue = true;
             continue;
           }
         }
-
-        if (hasValue) {
```

Comment:
This is changing behavior no?

Reply:
The earlier `continue` continues the inner loop, the `hasValue` is trying to detect if there is a value in the inner loop and the check later skips over adding the current outer iteration to the `missing` array

Reply:
Hm.  Yes, you're right.  Sonar didn't think so, but you're right.

I will revert.